### ISSUE TITLE
gfx_ScaleSprite doesn't seem to output correct data

### ISSUE THREAD

**Issue by Eeems** [CE-Programming/toolchain#264]
`gfx_ScaleSprite` + `gfx_Sprite_NoClip` output:
![image](https://user-images.githubusercontent.com/1176979/84581618-6e973f80-ada0-11ea-9c9b-ff73d043874a.png)
`gfx_ScaledSprite_NoClip` output:
![image](https://user-images.githubusercontent.com/1176979/84581621-7ce55b80-ada0-11ea-8065-d163cda7a2d5.png)

It's entirely possible that I'm doing this wrong. Here is the code I'm using to do the resize:
```c
void Sprite_resize(struct Sprite* self, int width, int height){
    gfx_sprite_t* resized_data = gfx_MallocSprite(width, height);
    gfx_ScaleSprite(self->data, resized_data);
    free(self->data);
    self->data = resized_data;
}

float scale = 3.0;
Sprite_resize(sprite, sprite->data->width * scale, sprite->data->height * scale);
```
`sprite->data` contains the gfx_sprite_t pointer.

It looks to me like it's clipping off the last row and column.

**Comment by Eeems** [comment]
Oh, and for reference, here is the unscaled original image
![bm1](https://user-images.githubusercontent.com/1176979/84585033-7d91e800-adc8-11ea-9fe5-26c4c9187812.png)

**Comment by mateoconlechuga** [comment]
Idk how that code even works. Can't you just do it the normal way?

**Comment by Eeems** [comment]
I am currently using gfx_ScaledSprite_NoClip.

That said, this method is available and documented. It should be either fixed or removed. Unless you are telling me that I'm doing it wrong?

**Comment by mateoconlechuga** [comment]
I'm telling you you are doing it wrong

**Comment by adriweb** [comment]
Well we have one use in the toolchain, here: https://github.com/CE-Programming/toolchain/blob/v8.8/examples/graphx/sprites_scaled/src/main.c#L20.
I suppose it works correctly there, unless it's by chance, but I guess someone would have noticed by now?

**Comment by Eeems** [comment]
> I'm telling you you are doing it wrong

Okay and how should it actually be used?

> Well we have one use in the toolchain, here: https://github.com/CE-Programming/toolchain/blob/v8.8/examples/graphx/sprites_scaled/src/main.c#L20.
> I suppose it works correctly there, unless it's by chance, but I guess someone would have noticed by now?

`gfx_ScaledSprite_NoClip` works fine, it's `gfx_ScaleSprite` that's the issue.

**Comment by Eeems** [comment]
So, any followup?

**Comment by mateoconlechuga** [comment]
So after spending way to much time on this, I've deduced that it isn't an actual bug. You might notice that extra pixels are added to the edge of the left side in the scaled image. This is because of the way sampling of pixels works for scaling. If you want a scale that works on pixel boundaries rather than sub-pixel sampling, I might recommend just iterating over the pixel data itself and creating the scaled image manually. Or using a combination of `gfx_ScaledSprite` / `gfx_GetSprite`.

I wish that I was able to resolve this more, but alas they are two different functions with different methods of operation. :/
